### PR TITLE
Cleanup GitHub actions

### PR DESCRIPTION
**Pull request by tautschnig**
Please see individual commit messages.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: .github/workflows/build-and-test-Linux.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ on:
       - '**'
 
 jobs:
+  # This job takes approximately 18 minutes
```
Comment (NlightNFotis) [inline .github/workflows/build-and-test-Linux.yaml:9]: I understand that these are informational only, but how are we going to track any changes to these numbers?

Or is the plan to be more laid back and only update them if we casually observe any change?

File: .github/workflows/build-and-test-Linux.yaml [hunk 2]
```diff
@@ -41,7 +41,7 @@ jobs:
       - name: Build CBMC tools
         run: |
           make -C src minisat2-download
-          make -C src CXX='ccache /usr/bin/g++' cbmc.dir goto-cc.dir goto-diff.dir -j2
```
Comment (NlightNFotis) [inline .github/workflows/build-and-test-Linux.yaml:43]: Why did the compiler change here?

**Review by NlightNFotis — APPROVED** [review]
LGTM, some minor questions.

File: .github/workflows/build-and-test-Linux.yaml (hunk 1 above)
Comment (tautschnig) [reply .github/workflows/build-and-test-Linux.yaml:9]: For now I was thinking keep this informal, but maybe we should go a little further and do actual checks? Don't know, I'm open to suggestions.

File: .github/workflows/build-and-test-Linux.yaml (hunk 2 above)
Comment (tautschnig) [reply .github/workflows/build-and-test-Linux.yaml:43]: This is to make sure we can actually use the compiler cache (and don't trash it by putting a different configuration in there).

**Pull request merged**